app/lib/advanced_search_finder_api.rb

```diff
@@ -0,0 +1,45 @@
+class AdvancedSearchFinderApi < FinderApi
+  TAXON_SEARCH_FILTER = "taxons".freeze
+
+  attr_reader :content_item
+
+  def content_item_with_search_results
+    filter_params["part_of_taxonomy_tree"] = taxon["content_id"]
+
+    content_item = fetch_content_item
+    search_response = fetch_search_response(content_item)
+    content_item = augment_content_item_links_with_taxon(content_item, taxon)
+    augment_content_item_with_results(content_item, search_response)
+  end
+
+  def taxon
+    # TODO: Rescue 404s
+    @taxon ||= Services.content_store.content_item(filter_params["part_of_taxonomy_tree"])
+  end
+
+private
+
+  def augment_content_item_links_with_taxon(content_item, taxon)
+    content_item["links"]["taxons"] = [taxon]
+    content_item
+  end
+
+  def augment_facets_with_dynamic_values(content_item, _)
+    augment_facets_with_dynamic_subgroups(content_item) if supergroups.any?
+  end
+
+  def augment_facets_with_dynamic_subgroups(content_item)
+    subgroups = supergroups.map(&:subgroups_as_hash).flatten
+    facet = find_facet(content_item, "content_purpose_subgroup")
```

Comment:
Should we guard against this not finding the facet?

Reply:
It's unlikely to be missing as it's configured in the content item, but now I've said that it's bound to go missing so I've short-cut this method if not present (next line)